force-app/main/default/classes/ServiceDeliveryDomain.cls

```diff
@@ -0,0 +1,16 @@
+/*
+ *
+ *  * Copyright (c) 2020, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+
+public with sharing class ServiceDeliveryDomain {
+    public void upsertServiceDeliveries(
+        List<ServiceDelivery__c> serviceDeliveriesToUpsert
+    ) {
+        upsert serviceDeliveriesToUpsert;
```

Comment:
I think we need to split out insert and update and then check permissions based on if either list has elements. Also we need to make sure they have field access.

Reply:
I am ok with checking both, I think it'd be pretty rare for someone to have one or another and be working in the attendance UI. On the question in slack, yes I believe we need to validate field permissions as well because this is an auraenabled method (from the controller). We just throw an error even on the invisible fields because the user should have perms to them.